### PR TITLE
[SERVER UI] FTP Parsing Workaround

### LINKED ISSUE #1338: Cannot preview artifacts in UI when uploaded to remote machine via FTP
### System information
- **Have I written custom code (as opposed to using a stock example script provided in MLflow)**:
No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Linux Ubuntu 18.04
- **MLflow installed from (source or binary)**: 
installed via from pypi via pip
- **MLflow version (run ``mlflow --version``)**:
0.9.1
- **Python version**: 
3.6.7
- **npm version (if running the dev UI):
- **Exact command to reproduce**:
Server runs with:
```
mlflow server --backend-store-uri ./mlruns --default-artifact-root ./artifacts -h 0.0.0.0 -p 5020
```
within a folder holding a `mlruns` and a `artifacts` folder

### Describe the problem
I am running the server on a remote machine.
On the local machine which uploads via the tracking API the
tracking uri is set to: `http://path.to.server:port`
and artifact uri to: `ftp://anonymous:@path.to.server/artifacts`

uploading the tracking information and artifacts work fine and everything shows in the ui:
image

However when I try to click one of the artifacts I always get the message that something went wrong:
image

Previewing the artifacts work when I store them locally on my machine but as soon as I transfer them via FTP and then try to preview them from the remote machine I get the error shown above.

### Source code / logs
-

### PR DESCRIPTION
**Pull request by nuno-chicoria**
## What changes are proposed in this pull request?

The ftp.nlst() function at line 106 creates a list that starts with "." and "..". These two items are not objects so, when accessing through FTP, MLflow throws an error about not being able to access those files. In this workaround, we iterate the list starting at index 2.

Fixes #1338 

## How is this patch tested?

TBD

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

Fixes bug where artifacts could not be visualised when storing them through an FTP connection on a remote server.

### What component(s) does this PR affect?

- [X] UI
- [ ] CLI
- [ ] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [X] Tracking
- [ ] Projects
- [X] Artifacts
- [ ] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [X] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/store/artifact/ftp_artifact_repo.py [hunk 1]
```diff
@@ -105,7 +105,7 @@ def list_artifacts(self, path=None):
                 return []
             artifact_files = ftp.nlst(list_dir)
             infos = []
-            for file_name in artifact_files:
+            for file_name in artifact_files[2:]:
```
Comment (mparkhe) [inline mlflow/store/artifact/ftp_artifact_repo.py]: Starting from element # 2 sounds dangerous. Can we instead skip if `file_name` is `"."` or `".."`

Comment (nuno-chicoria) [reply mlflow/store/artifact/ftp_artifact_repo.py]: Thank you for the input. I removed the `[2:]` and added a `list(filter(lambda x: x != "." and x != "..", artifact_list))` before to remove any `"."` and `".."`.

Comment (nuno-chicoria) [reply mlflow/store/artifact/ftp_artifact_repo.py]: Hi @mparkhe! Could you give us a follow up on this? Do you think it is relevant to change and merge with MLflow master? Is there anything we can do to help more with the process?

**Review by mparkhe — APPROVED** [review]

**Comment by mparkhe** [comment]
fixes #1814 #1338

**Pull request merged**